libultraship/libultraship/HIDController.cpp

```diff
@@ -0,0 +1,369 @@
+                    }
+
+                    if (StickDeadzone != 0 && StickDeadzone != AxisThreshold) {
+                        SPDLOG_TRACE("Invalid Deadzone configured. Left/Right was {} and Up/Down is {}", StickDeadzone, AxisThreshold);
+                    }
+
+                    StickDeadzone = AxisThreshold;
+                    StickAxisX = Axis;
+                }
+
+                if (NegButton == BTN_STICKUP || NegButton == BTN_STICKDOWN) {
+                    if (StickAxisY != HID_CONTROLLER_AXIS_INVALID && StickAxisY != Axis) {
+                        SPDLOG_TRACE("Invalid NegStickY configured. Pos was {} and Neg is {}", StickAxisY, Axis);
+                    }
+
+                    if (StickDeadzone != 0 && StickDeadzone != AxisThreshold) {
+                        SPDLOG_TRACE("Invalid Deadzone misconfigured. Left/Right was {} and Up/Down is {}", StickDeadzone, AxisThreshold);
+                    }
+
+                    StickDeadzone = AxisThreshold;
+                    StickAxisY = Axis;
+                }
+            }
+
+            if (StickAxisX != HID_CONTROLLER_AXIS_INVALID && StickAxisY != HID_CONTROLLER_AXIS_INVALID) {
+                auto AxisValueX = SDL_JoystickGetAxis(Cont, StickAxisX);
+                auto AxisValueY = SDL_JoystickGetAxis(Cont, StickAxisY);
+                NormalizeStickAxis(AxisValueX, AxisValueY, StickDeadzone);
+            }
+        }
+    }
+
+    void HIDController::WriteToSource(ControllerCallback* controller)
+    {
+        if (SDL_JoystickHasRumble(Cont)) {
+            if (controller->rumble > 0) {
+                SDL_JoystickRumble(Cont, 0xFFFF * Game::Settings.controller.rumble_strength, 0xFFFF * Game::Settings.controller.rumble_strength, 1);
+            }
+        }
+
+        if (SDL_JoystickHasLED(Cont)) {
+            if (controller->ledColor == 1) {
+                SDL_JoystickSetLED(Cont, 255, 0, 0);
+            }
+            else {
+                SDL_JoystickSetLED(Cont, 0, 255, 0);
+            }
+        }
+    }
+
+    void HIDController::CreateDefaultBinding() {
+        std::string ConfSection = GetBindingConfSection();
+        std::shared_ptr<ConfigFile> pConf = GlobalCtx2::GetInstance()->GetConfig();
+        ConfigFile& Conf = *pConf.get();
+
+        Conf[ConfSection][STR(BTN_CRIGHT)] = std::to_string(9);
+        Conf[ConfSection][STR(BTN_CLEFT)] = std::to_string(8);
+        Conf[ConfSection][STR(BTN_CDOWN)] = std::to_string(7);
+        Conf[ConfSection][STR(BTN_CUP)] = std::to_string(6);
+        //Conf[ConfSection][STR(BTN_CRIGHT + "_2")] = std::to_string(SDL_CONTROLLER_BUTTON_X);
+        //Conf[ConfSection][STR(BTN_CLEFT + "_2")] = std::to_string(SDL_CONTROLLER_BUTTON_Y);
+        //Conf[ConfSection][STR(BTN_CDOWN + "_2")] = std::to_string(SDL_CONTROLLER_BUTTON_RIGHTSHOULDER);
+        //Conf[ConfSection][STR(BTN_CUP + "_2")] = std::to_string(SDL_CONTROLLER_BUTTON_RIGHTSTICK);
```

Comment:
Consider removing these comments if they are not used.

Reply:
Resolved in [f7caeb5](https://github.com/HarbourMasters/Shipwright/pull/86/commits/f7caeb51d55dd8922a8d22321e1e4d0ff36f276e)